### DISCUSSION TITLE
升级3.7.1/2后对象存储无法使用

### DISCUSSION THREAD

**Discussion by Meeeeeet6** [AlistGo/alist#2860] [Invalid question]
在升级Alist V3.7.1开始出现这种情况。升级到V3.7.2之后仍然存在，无解。对象存储供应商为B2和Cloudflare R2，均出现此问题；数据库为postgres

情况描述：
1.在Alist中打开对象存储之后，报错如下（如图）：
failed get link: InvalidParameter: 1 validation error(s) found. - minimum field size of 1, GetObjectInput.Key.
![image](https://user-images.githubusercontent.com/74605260/210161939-970ec133-a002-4c1b-8ae0-c0b5683afd2b.png)

2.如果只是打开目录，不是打开文件，会一直循环重复的打开一个目录，如图：
![image](https://user-images.githubusercontent.com/74605260/210161961-1870f884-fa53-4aa9-83dc-20fc0f7388b3.png)

其他存储暂未发现影响。

**Comment by Meeeeeet6** [comment]
发现演示站中B2存储也出现了无法访问的情况：
链接如下：https://al.nn.ci/b2
![image](https://user-images.githubusercontent.com/74605260/210161984-f8d82aa0-8dda-43d6-b319-bbf7685fb8bb.png)

**Comment by DarremMolko** [comment]
For the the second problem you have to edit your S3 driver and fill the "root folder path" field (introduced in v3.7.2):
![Screenshot_20230101-171037](https://user-images.githubusercontent.com/70179709/210183639-f1542a5f-e3fb-4ca8-a540-cb99f017a225.jpg)

This worked for me with idrive e2, Synology C2, and ibm COS, but for Backblaze B2 I also had to enable "Force path style":
![Screenshot_20230101-171056](https://user-images.githubusercontent.com/70179709/210183500-31085128-a25d-45f6-ad42-6a249ef23adb.jpg)

**Reply by Meeeeeet6** [reply]
Hello, thank you very much for your help. My problem has been solved successfully!But I still have a question to ask you. Why should "forced path style" be enabled for B2

**Reply by DarremMolko** [reply]
Last time I tried I had to enable it because filling the "root folder path" wasn't enough to fix the problem from your second screenshot.

**Reply by Meeeeeet6** [reply]
> Last time I tried I had to enable it because filling the "root folder path" wasn't enough to fix the problem from your second screenshot.

Is that what you're talking about?
![image](https://user-images.githubusercontent.com/74605260/210189857-333c3f14-ad47-424a-b7e1-4d33b479a2a9.png)

To solve this problem, I tried to mount B2. After setting the root path, the problem did not occur.
![image](https://user-images.githubusercontent.com/74605260/210189870-d9950863-b224-4100-acd1-b93afd48d2f4.png)

**Reply by Meeeeeet6** [reply]
> > Last time I tried I had to enable it because filling the "root folder path" wasn't enough to fix the problem from your second screenshot.
> 
> Is that what you're talking about? ![image](https://user-images.githubusercontent.com/74605260/210189857-333c3f14-ad47-424a-b7e1-4d33b479a2a9.png)
> 
> To solve this problem, I tried to mount B2. After setting the root path, the problem did not occur. ![image](https://user-images.githubusercontent.com/74605260/210189870-d9950863-b224-4100-acd1-b93afd48d2f4.png)

并没有设置”强制路径样式“

**Reply by DarremMolko** [reply]
I'm glad to hear it worked for you, maybe it didn't for me because of the browser cache.
